gateway/ln-gateway/src/federation_manager.rs

```diff
@@ -127,4 +127,19 @@ impl FederationManager {
     pub async fn has_federation(&self, federation_id: FederationId) -> bool {
         self.clients.read().await.contains_key(&federation_id)
     }
+
+    pub async fn set_next_scid(&self, next_scid: u64) {
+        *self.next_scid.lock().await = next_scid;
+    }
```

Comment:
Would need more refactoring, but feels like that should be set by the constructor.

Reply:
Added a TODO comment